lib/css_parser/rule_set.rb

```diff
@@ -22,90 +24,152 @@ class RuleSet
+          @order = order
+        end
+
+        def value=(value)
+          value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
+          raise ArgumentError, 'value is empty' if value.empty?
+
+          @value = value.freeze
+        end
+
+        def to_s
+          return value unless important
 
-      def initialize(value, important: nil, order: 0)
-        self.value = value
-        @important = important.nil? ? !value.match(CssParser::IMPORTANT_IN_PROPERTY_RX).nil? : important
-        @order = order
+          "#{value} !important"
+        end
       end
 
-      def value=(value)
-        value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
-        raise ArgumentError, 'value is empty' if value.empty?
+      attr_reader :declarations
+      private :declarations
 
-        @value = value.freeze
+      def initialize(declarations = {})
+        @order = 0
+        @declarations = {}
+        declarations.each { |property, value| add_declaration!(property, value) }
       end
 
-      def to_s
-        return value unless important
+      # Add a CSS declaration
+      # @param [#to_s] property that should be added
+      # @param [Value, #to_s] value of the property
+      #
+      # @example
+      #   declarations['color'] = 'blue'
+      #
+      #   puts declarations['color']
+      #   => #<CssParser::RuleSet::Declarations::Value:0x000000000305c730 @important=false, @order=1, @value="blue">
+      #
+      # @example
+      #   declarations['margin'] = '0px auto !important'
+      #
+      #   puts declarations['margin']
+      #   => #<CssParser::RuleSet::Declarations::Value:0x00000000030c1838 @important=true, @order=2, @value="0px auto">
+      #
+      # If the property already exists its value will be over-written.
+      # If the value is empty - property will be deleted
+      def []=(property, value)
+        property = property.to_s.downcase
+        property.strip!
+
+        if value.is_a?(Value)
+          declarations[property] = value
+          return
+        end
+
+        if value.to_s.strip.empty?
+          remove_declaration!(property)
+          return
+        end
+
+        declarations[property] = Value.new(value, order: @order += 1)
+      end
+      alias add_declaration! []=
+
+      def [](property)
+        property = property.to_s.downcase
+        property.strip!
+        declarations[property]
+      end
+      alias get_value []
+
+      def key?(property)
+        property = property.to_s.downcase
+        property.strip!
+        declarations.key?(property)
+      end
+
+      def size
+        declarations.size
+      end
+
+      # Remove CSS declaration
+      # @param [#to_s] property property to be removed
+      #
+      # @example
+      #   declarations.remove_declaration!('color')
+      def remove_declaration!(property)
```

Comment:
would be neat to make this consistent with the hashy interface as `def delete`

Reply:
Done.